ccm_web/client/src/hooks/useGlobals.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from 'react'
+
+import usePromise from './usePromise'
+import * as api from '../api'
+import { Globals } from '../models'
+
+/* Hook for fetching global data and checking whether user is authenticated. */
+function useGlobals (ltiKey: string | undefined): [
+  Globals | undefined,
+  boolean | undefined,
+  boolean,
+  Error | undefined
+] {
+  const [globals, setGlobals] = useState(undefined as Globals | undefined)
+
+  const getGlobals = async (): Promise<Globals> => await api.getGlobals(ltiKey)
+  const [doGetGlobals, getGlobalsLoading, getGlobalsError] = usePromise(
+    getGlobals,
+    (value: Globals) => setGlobals(value)
+  )
+  useEffect(() => {
+    /* eslint-disable @typescript-eslint/no-floating-promises */
+    doGetGlobals()
+  }, [])
```

Comment:
is this useEffect is making sure that the globals is fetched just once?

Reply:
Yes.